sway/tree/view.c

```diff
@@ -785,9 +785,21 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
 
 	view_execute_criteria(view);
 
+
+#if HAVE_XWAYLAND
```

Comment:
I feel like the logic could be simplified here, perhaps with something like:

```c
bool set_focus = should_focus(view);
if (wlr_surface_is_xwayland_surface(wlr_surface)) {
    set_focus = set_focus || wlr_xwayland_or_surface_wants_focus(…);
}
if (set_focus) {
    input_manager_set_focus(…);
}
```

Reply:
Yep! Agree.